tools/ci_build/github/linux/docker/scripts/install_dotnet.sh

```diff
@@ -2,7 +2,7 @@
 set -e -x
 
 if [ -f /etc/redhat-release ]; then
-    dnf update --refresh \
+    dnf update --refresh -y \
```

Comment:
Do not call "dnf update". Sometimes the base image we got is a mess. When we run "dnf update" it could

1. Upgrade the existing packages to incompatible major versions. From example, we got a CUDA 11.x base image, then we run "dnf update" and some CUDA libs become to CUDA 12.x.  It happened a lot.

2. It could just fail to run because of package conflicts. For example, the ROCM's CentOS8 image has two AMD yum repos that conflict with each other. We still could install package from them but whenever we run `dnf update` it will become to a mess. 

On the other hand, For Ubuntu we always need to call `apt-get update` right before any `apt-get install`. dnf's update command means "upgrade", it is not the same thing as "apt-get update".